crates/turbopack-ecmascript/src/tree_shake/cjs_finder.rs

```diff
@@ -1,11 +1,146 @@
+                    with, specifiers, ..
+                })) => {
+                    if let Some(with) = with.as_deref().and_then(|v| v.as_import_with()) {
+                        for item in with.values.iter() {
+                            if item.key.sym == *TURBOPACK_HELPER {
+                                // Skip tree shaking if the import is from turbopack-helper
+                                return true;
+                            }
+                        }
+                    }
+
+                    // Tree shaking has a bug related to ModuleExportName::Str
+                    for s in specifiers.iter() {
+                        if let ImportSpecifier::Named(is) = s {
+                            if matches!(is.imported, Some(ModuleExportName::Str(..))) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                // Tree shaking has a bug related to ModuleExportName::Str
+                ModuleItem::ModuleDecl(ModuleDecl::ExportNamed(NamedExport {
+                    src: Some(..),
+                    specifiers,
+                    ..
+                })) => {
+                    for s in specifiers {
+                        if let ExportSpecifier::Named(es) = s {
+                            if matches!(es.orig, ModuleExportName::Str(..))
+                                || matches!(es.exported, Some(ModuleExportName::Str(..)))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                // Skip sever actions
+                ModuleItem::Stmt(Stmt::Expr(ExprStmt {
+                    expr: box Expr::Lit(Lit::Str(Str { value, .. })),
+                    ..
+                })) => {
+                    if value == "use server" {
+                        return true;
+                    }
+                }
+
+                // Skip special reexports that are recognized by next.js
+                ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(ExportDecl {
+                    decl: Decl::Var(box VarDecl { decls, .. }),
+                    ..
+                })) => {
+                    for decl in decls {
+                        if let Pat::Ident(name) = &decl.name {
+                            if is_next_js_special_export(&name.sym) {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                // Skip special reexports that are recognized by next.js
+                ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(ExportDecl {
+                    decl: Decl::Fn(f),
+                    ..
+                })) => {
+                    if is_next_js_special_export(&f.ident.sym) {
+                        return true;
+                    }
+                }
```

Comment:
I don't love the idea of putting more highly-nextjs-specific things in the core parts of turbopack. Could we instead make this configurable (e.g. with a passed in closure) and move `is_next_js_special_export` into the nextjs PR?